pkg/virtual/workspaces/registry/rest.go

```diff
@@ -628,41 +443,17 @@ func (s *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 	}
 
 	// Then create the owner role related to the given workspace.
-	// Note that ResourceNames contains the workspace pretty name for now.
-	// It will be updated later on when the internal name of the workspace is known.
 	ownerClusterRole := createClusterRole(ownerRoleBindingName, workspace.Name, OwnerRoleType)
 	if _, err := s.kubeClusterClient.Cluster(orgClusterName).RbacV1().ClusterRoles().Create(ctx, ownerClusterRole, metav1.CreateOptions{}); err != nil && !kerrors.IsAlreadyExists(err) {
 		return nil, kerrors.NewForbidden(tenancyv1beta1.Resource("workspaces"), workspace.Name, err)
 	}
 
-	// Then try to create the workspace object itself, first with the pretty name,
-	// retrying with increasing suffixes until a workspace with the same name
-	// doesn't already exist.
-	// The suffixed name based on the pretty name will be the internal name
-	clusterWorkspace := &tenancyv1alpha1.ClusterWorkspace{
-		ObjectMeta: workspace.ObjectMeta,
-		Spec: tenancyv1alpha1.ClusterWorkspaceSpec{
-			Type: workspace.Spec.Type,
-		},
-	}
-	createdClusterWorkspace, err := s.kcpClusterClient.Cluster(orgClusterName).TenancyV1alpha1().ClusterWorkspaces().Create(ctx, clusterWorkspace, metav1.CreateOptions{})
-	if err != nil && kerrors.IsAlreadyExists(err) {
-		clusterWorkspace.Name = ""
-		clusterWorkspace.GenerateName = workspace.Name + "-"
-		createdClusterWorkspace, err = s.kcpClusterClient.Cluster(orgClusterName).TenancyV1alpha1().ClusterWorkspaces().Create(ctx, clusterWorkspace, metav1.CreateOptions{})
-	}
-	if err != nil {
-		_ = s.kubeClusterClient.Cluster(orgClusterName).RbacV1().ClusterRoles().Delete(ctx, ownerClusterRole.Name, metav1.DeleteOptions{GracePeriodSeconds: &zero})
-		_ = s.kubeClusterClient.Cluster(orgClusterName).RbacV1().ClusterRoleBindings().Delete(ctx, clusterRoleBinding.Name, metav1.DeleteOptions{GracePeriodSeconds: &zero})
-		return nil, err
-	}
-
 	// Update the cluster roles with the new workspace internal name, and also
 	// add the internal name as a label, to allow searching with it later on.
 	for i := range ownerClusterRole.Rules {
 		ownerClusterRole.Rules[i].ResourceNames = []string{createdClusterWorkspace.Name}
```

Comment:
Now that we're not potentially creating a workspace with a name that differs from the user's request, we should be able to consolidate the ClusterRole to a single Create call, instead of Create+Update, right?

Reply:
You're right. I missed it, but this obviously needs more cleanup.

Reply:
Additional cleanup done in commit ab6cf45a76494764713e1bd66e27413c22c7f29f